Bearded.Utilities/Noise/ProceduralTexture.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+
+namespace Bearded.Utilities.Noise
+{
+    public static class ProceduralTexture
+    {
+        public static IProceduralTexture FromArray(double[,] array, IInterpolationMethod2 interpolation) =>
+            new ArrayProceduralTexture(array, interpolation);
+
+        private sealed class ArrayProceduralTexture : IProceduralTexture
+        {
+            private readonly int width;
+            private readonly int height;
+            private readonly double[,] array;
+            private readonly IInterpolationMethod2 interpolation;
+
+            public ArrayProceduralTexture(double[,] array, IInterpolationMethod2 interpolation)
+            {
+                width = array.GetLength(0);
+                height = array.GetLength(1);
+                this.array = new double[width, height];
+                Array.Copy(array, this.array, array.Length);
+
+                this.interpolation = interpolation;
+            }
+
+            public double ValueAt(double x, double y)
+            {
+                if (x < 0 || x >= 1)
```

Comment:
See my arguments above. I want to keep these for early failure, instead of presenting the failure with something that's really not useful without knowing how it relates back to the original input parameters.